eduHub/src/main/kotlin/com/arsiu/eduhub/controller/nats/assignment/AssignmentNatsControllerFindAll.kt

```diff
@@ -20,11 +21,11 @@ class AssignmentNatsControllerFindAll(
     override val subject: String = ASSIGNMENT_FIND_ALL
     override val parser: Parser<FindAllAssignmentRequest> = FindAllAssignmentRequest.parser()
 
-    override fun handler(request: FindAllAssignmentRequest): FindAllAssignmentResponse =
-        runCatching {
-            getSuccessResponse()
-        }.getOrElse { ex ->
-            getFailureResponse(ex.javaClass.simpleName, ex.toString())
+    override fun handler(request: FindAllAssignmentRequest): Mono<FindAllAssignmentResponse> =
+        Mono.defer {
+            Mono.just(getSuccessResponse())
```

Comment:
your getSuccessResponse uses `block` inside. so you don't have any async non-blocking features. refactor it to make truly reactive

Reply:
`+`